### PR TITLE
Update pubnub index.d.ts - HistoryMessage.meta entries should be any type

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
The history message meta object should be `any` considering meta objects are not flat

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Example code:

```js
await pubnub.publish({
	channel,
	message,
	meta: {
		user: {
			name: user.name,
			profilePicture: user.profilePicture
		}
	}
})

// ---

pubnub.history({
	channel: chatChannel,
	count: 100,
	includeMeta: true,
}, (status, response) => {
	if (status.statusCode === 200) {
		setMessages(response.messages.map(message => ({
			message: message.entry,
			user: message.meta.user
		})))
	}
})
```

### CODE REVIEW

File: types/pubnub/index.d.ts [hunk 1]
```diff
@@ -468,7 +468,7 @@ declare namespace Pubnub {
         entry: any;
         timetoken?: string | number;
         meta?: {
-            [key: string]: string;
+            [key: string]: any;
```
Comment (FlorianDr) [inline types/pubnub/index.d.ts:471]: Hey! Thank you for the change. I was just having a look into the flow annotation: 

https://github.com/pubnub/javascript/blob/master/src/core/flow_interfaces.js#L246-L250

Shall we follow and make it ``meta?: object``?

Comment (Ethan-Arrowood) [reply types/pubnub/index.d.ts:471]: Yeah that seems reasonable to me

**Review by FlorianDr — APPROVED** [review]
LGTM!

**Review by orta — APPROVED** [review]

**Comment by Ethan-Arrowood** [comment]
Ready to merge

**Pull request merged**